src/components/breadcrumbs/BreadcrumbLink.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import { forwardRef } from 'react';
+import { truncate } from '~common/helpers/styles';
+import { LinkHighlight } from '../links';
+import { LinkStandalone, LinkStandaloneProps } from '../links/LinkStandalone';
+
+type LinkProps = Omit<LinkStandaloneProps, 'children' | 'isDisabled'> & {
+  linkElement: React.ReactNode;
+};
+
+export type BreadcrumbLinkProps = (
+  | (LinkProps & { isChild?: never })
+  | (Omit<LinkProps, 'to'> & { isChild: true; to?: '' })
+) & {
+  hasEllipsis?: boolean;
+};
+
+const BreadcrumbLinkContainer = styled.span`
+  font-size: var(--echoes-font-size-10);
+  line-height: var(--echoes-line-height-10);
+  max-width: var(--echoes-sizes-breadcrumbs-max-width-default);
+
+  ${truncate}
+`;
+
+const BreadcrumbLinkBase = forwardRef<HTMLAnchorElement, BreadcrumbLinkProps>((props, ref) => {
+  const { hasEllipsis, isChild, linkElement, to = '', ...linkProps } = props;
+
+  const title = hasEllipsis ? (linkElement as string) : undefined;
+
+  if (isChild) {
+    return (
+      <LinkStandalone {...linkProps} isDisabled ref={ref}>
+        {linkElement}
+      </LinkStandalone>
+    );
+  }
+
+  const Link = (
+    <LinkStandalone
+      {...linkProps}
+      highlight={LinkHighlight.Subdued}
+      ref={ref}
+      title={title}
+      to={to}>
+      {linkElement}
+    </LinkStandalone>
+  );
+
+  return hasEllipsis ? <BreadcrumbLinkContainer>{Link}</BreadcrumbLinkContainer> : Link;
+});
+
+BreadcrumbLinkBase.displayName = 'BreadcrumbLinkBase';
+
+export const BreadcrumbLink = styled(BreadcrumbLinkBase)`
+  font: var(--echoes-typography-text-default-regular);
+  font-size: var(--echoes-font-size-10);
+  line-height: var(--echoes-line-height-10);
```

Comment:
Can't you use `echoes-typography-text-small-regular` which already contains size-10 and line-height-10 instead of using another typography and changing it's size?

Also if this is set on the BreadcrumbList then it should be inherited by all links inside.